### PR TITLE
Implement more flows ops methods part 2

### LINKED ISSUE #67: implement more flows ops methods - part 2
Implement methods:
* interleave
* mapConcat
* mapPar
* mapParUnordered
* sliding
* alsoTo
* alsoToTap

### PR DESCRIPTION
**Pull request by emil-bar**
Closes #67

### CODE REVIEW

**Comment by adamw** [comment]
I assume these are directly translated from ox - any places where the implementations diverge?

**Comment by uini223** [comment]
> I assume these are directly translated from ox - any places where the implementations diverge?

 Yes, all of them are translated, including tests (I rely on them to see if implementation works). 
 
 One additional thing is that for methods that use `Scope.unsupervised` or `Scope.supervised` I need to catch ExecutionException and rethrow it for the tests to pass. See: https://github.com/softwaremill/jox/pull/75/files#diff-46d2537d747d24775cf70630d584ac42326ea6902d3ddd10f5a861841c2c3b6cR738

File: flows/src/main/java/com/softwaremill/jox/flows/Flows.java [hunk 1]
```diff
@@ -222,4 +230,82 @@ public static <T> Flow<T> failed(Exception t) {
...
+     *
+     * @param flows
+     *   The flows whose elements will be interleaved.
+     * @param segmentSize
+     *   The number of elements sent from each flow before switching to the next one.
+     * @param eagerComplete
+     *   If `true`, the returned flow is completed as soon as any of the flows completes. If `false`, the interleaving continues with the
+     *   remaining non-completed flows.
+     */
+    public static <T> Flow<T> interleaveAll(List<Flow<T>> flows, int segmentSize, boolean eagerComplete, int bufferCapacity) {
```
Comment (uini223) [inline flows/src/main/java/com/softwaremill/jox/flows/Flows.java:252]: I must have created this method before adding scope value for buffer capacity. I'll overload the method in the next PR

**Comment by adamw** [comment]
> One additional thing is that for methods that use Scope.unsupervised or Scope.supervised I need to catch ExecutionException and rethrow cause for the tests to pass. See: https://github.com/softwaremill/jox/pull/75/files#diff-46d2537d747d24775cf70630d584ac42326ea6902d3ddd10f5a861841c2c3b6cR738

Are these tests which verify what exception was thrown? Maybe we could instead change the tests to verify that an EE was thrown?

EE are something that we might want to change in general (see https://github.com/softwaremill/jox/issues/59), but until then, it would be good to keep consistency.

**Comment by uini223** [comment]
> > One additional thing is that for methods that use Scope.unsupervised or Scope.supervised I need to catch ExecutionException and rethrow cause for the tests to pass. See: https://github.com/softwaremill/jox/pull/75/files#diff-46d2537d747d24775cf70630d584ac42326ea6902d3ddd10f5a861841c2c3b6cR738
> 
> Are these tests which verify what exception was thrown? Maybe we could instead change the tests to verify that an EE was thrown?
> 
> EE are something that we might want to change in general (see #59), but until then, it would be good to keep consistency.

Yes, those are the tests that verify it. I can fix the tests and verify that EE is thrown. Overall those exceptions are ugly as we are adding 2/3 wrappers on top of source exception in most cases

**Comment by adamw** [comment]
@uini223 let's get rid of them in a separate issue then :)

**Comment by emil-bar** [comment]
> @uini223 let's get rid of them in a separate issue then :)

Removed all rethrows that were present in the code I've added recently

**Pull request merged**